src/main/java/pl/umk/mat/git2befit/service/workout/factory/implementation/FBWTrainingPlan.java

```diff
@@ -9,101 +9,108 @@
 import pl.umk.mat.git2befit.repository.workout.ExerciseRepository;
 
 import java.util.*;
+import java.util.function.BooleanSupplier;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 @Component
 public class FBWTrainingPlan implements TrainingPlanInterface {
     private static final String TRAINING_TYPE = "FBW";
-    private static final List<String> bodyPartsList = List.of("SIXPACK", "CALVES", "BICEPS", "TRICEPS","SHOULDERS", "CHEST", "BACK", "THIGHS");
+    private static final List<String> bodyPartsList = List.of("SIXPACK", "CALVES", "BICEPS", "TRICEPS", "SHOULDERS", "CHEST", "BACK", "THIGHS");
+    private static final int ONE_DAY = 1;
+
     private final ExerciseRepository exerciseRepository;
+
+    private List<Exercise> exercisesWithoutEquipment = new ArrayList<>();
+    private List<Exercise> exercisesWithEquipment = new ArrayList<>();
     private TrainingForm trainingForm;
+    private TrainingForm localTrainingForm;
 
     public FBWTrainingPlan(ExerciseRepository exerciseRepository) {
         this.exerciseRepository = exerciseRepository;
     }
 
-    @Override
-    public List<Training> create(TrainingForm trainingForm) {
+    private void initialize(TrainingForm trainingForm) {
         this.trainingForm = trainingForm;
-        List<Exercise> exerciseListFilteredByTrainingType = getExerciseListFilteredByTrainingType();
-        List<Exercise> exercisesWithoutEquipment = exerciseRepository.getAllWithNoEquipmentForTrainingTypeName(TRAINING_TYPE);
-        List<Exercise> exercisesWithEquipment = filterAllByAvailableEquipment(exerciseListFilteredByTrainingType, trainingForm.getEquipmentIDs());
-        return assignExercisesToBodyPart(exercisesWithEquipment,exercisesWithoutEquipment);
+        this.localTrainingForm = trainingForm;
+
+        var exerciseListFilteredByTrainingType = exerciseRepository.getAllByTrainingTypes_Name(TRAINING_TYPE);
+        BooleanSupplier isContainsEquipmentWithout = () -> trainingForm.getEquipmentIDs().contains(20L);
+        if (isContainsEquipmentWithout.getAsBoolean()) {
+            localTrainingForm.getEquipmentIDs().remove(20L);
+            exercisesWithoutEquipment = exerciseRepository.getAllWithNoEquipmentForTrainingTypeName(TRAINING_TYPE);
+        }
+        exercisesWithEquipment = filterAllByAvailableEquipment(exerciseListFilteredByTrainingType, trainingForm.getEquipmentIDs());
+
+
+        Predicate<TrainingForm> checkIfScheduleTypeEqualsREPETITIVE = trainingForm1 -> trainingForm1.getScheduleType().equals("REPETITIVE");
```

Comment:
Czy tak powinno się serio robić? Nazwa tego predicate'a mówi dokładnie to samo co by mówiło to odwołanie. Może i jest długie, ale czytelność jest ta sama, a jednak się to trochę skraca.